salt/metalk8s/repo/installed.sls

```diff
@@ -19,27 +27,64 @@ Generate package repositories nginx configuration:
     - backup: false
     - defaults:
         listening_port: {{ repo.port }}
+        nginx_registry_config: {{ nginx_registry_config }}
+        nginx_registry: {{ nginx_registry }}
+
+Inject nginx image:
+  containerd.image_managed:
+    - name: docker.io/library/nginx:1.15.8
+    - archive_path: {{ metalk8s.iso_root_path }}/images/nginx-1.15.8.tar
 
-Install package repositories manifest:
+Install repositories manifest:
   metalk8s.static_pod_managed:
-    - name: /etc/kubernetes/manifests/package-repositories.yaml
-    - source: salt://{{ slspath }}/files/package-repositories-manifest.yaml.j2
+    - name: /etc/kubernetes/manifests/repositories.yaml
+    - source: salt://{{ slspath }}/files/repositories-manifest.yaml.j2
     - config_files:
-      - {{ nginx_configuration_path }}
+      - {{ nginx_default_conf }}
+      - {{ nginx_registry_path }}
+      - {{ nginx_registry_config_path }}
     - context:
         container_port: {{ repo.port }}
-        image: {{ package_repositories_image }}
-        name: {{ package_repositories_name }}
-        version: {{ package_repositories_version }}
+        image: docker.io/library/nginx:1.15.8
+        name: {{ repositories_name }}
+        version: {{ repositories_version }}
         packages_path: {{ metalk8s.iso_root_path }}/{{ repo.relative_path }}
-        nginx_configuration_path: {{ nginx_configuration_path }}
+        nginx_confd_path: {{ nginx_confd }}
+        images_path: {{ images_path }}
     - onchanges:
-      - file: Generate package repositories nginx configuration
+      - containerd: Inject nginx image
+      - file: Generate repositories nginx configuration
+      - file: Deploy container registry nginx configuration
+      - file: Generate container registry configuration
 
-Ensure package repositories container is up:
+Ensure repositories container is up:
   module.wait:
     - cri.wait_container:
-      - name: {{ package_repositories_name }}
+      - name: {{ repositories_name }}
       - state: running
     - require:
-      - metalk8s: Install package repositories manifest
+      - metalk8s: Install repositories manifest
+
+Deploy container registry nginx configuration:
```

Comment:
Shouldn't this happen before we deploy the service manifest? I'm aware the `onchanges` acts as a `require`, so things will be executed by Salt 'in the right order', but I think it also makes it easier to read/follow an SLS when tasks are 'in the expected order of execution' within the source.

Same applies to the step below.

Reply:
Done, I moved `Deploy container registry nginx configuration` and `Generate container registry configuration` before `Install repositories manifest`